src/java/compat/src/main/java/com/tencent/wcdb/compat/SQLiteDatabase.java

```diff
@@ -1008,13 +1008,17 @@ public void execSQL(String sql) {
      * @param bindArgs only byte[], String, Long and Double are supported in bindArgs.
      * @throws SQLException if the SQL string is invalid
      */
-    public void execSQL(String sql, Object[] bindArgs) {
-        execSQL(sql, bindArgs, null);
+    public int execSQL(String sql, Object[] bindArgs) {
+        return execSQL(sql, bindArgs, null);
     }
 
-    public void execSQL(String sql, Object[] bindArgs, CancellationSignal cancellationSignal) {
+    public int execSQL(String sql, Object[] bindArgs, CancellationSignal cancellationSignal) {
+        return execSQL(sql, bindArgs, cancellationSignal, false);
```

Comment:
You can simply use sql.toLowerCase().startWith("select") to check if the sql is read-only.